smartsim/_core/entrypoints/indirect.py

```diff
@@ -0,0 +1,228 @@
+    logger.debug("Indirect step starting")
+
+    with open(output_path, "w+", encoding="utf-8") as ofp, open(
+        error_path, "w+", encoding="utf-8"
+    ) as efp:
+        start_detail = f"Proxy process {proxy_pid}"
+        start_rc: t.Optional[int] = None
+
+        try:
+            process = psutil.Popen(
+                cleaned_cmd,
+                cwd=cwd,
+                stdout=ofp.fileno(),
+                stderr=efp.fileno(),
+                close_fds=True,
+            )
+            STEP_PID = process.pid
+            logger.info(f"Indirect step step {STEP_PID} started")
+            start_detail += f" started child process {STEP_PID}"
+
+        except Exception as ex:
+            start_detail += f" failed to start child process. {ex}"
+            start_rc = 1
+            logger.error("Failed to create process", exc_info=True)
+            cleanup()
+            return 1
+        finally:
+            track_event(
+                get_ts(),
+                proxy_pid,
+                "",  # step_id for unmanaged task is always empty
+                etype,
+                "start",
+                status_path,
+                logger,
+                detail=start_detail,
+                return_code=start_rc,
+            )
+
+        ret_code = process.wait()
+
+    logger.info(f"Indirect step {STEP_PID} complete")
+    msg = f"Process {STEP_PID} finished with return code: {ret_code}"
+    track_event(
+        get_ts(),
+        proxy_pid,
+        "",  # step_id for unmanaged task is always empty
+        etype,
+        "stop",
+        status_path,
+        logger,
+        detail=msg,
+        return_code=ret_code,
+    )
+    cleanup()
+
+    return ret_code
+
+
+def cleanup() -> None:
+    """Perform cleanup required for clean termination"""
+    global STEP_PID  # pylint: disable=global-statement
+    if STEP_PID is None:
+        return
+
+    try:
+        # attempt to stop the subprocess performing step-execution
+        if psutil.pid_exists(STEP_PID):
+            process = psutil.Process(STEP_PID)
+            process.terminate()
+
+    except psutil.NoSuchProcess:
+        # swallow exception to avoid overwriting outputs from cmd
+        ...
+
+    except OSError as ex:
+        logger.warning(f"Failed to clean up step executor gracefully: {ex}")
+    finally:
+        STEP_PID = None
+
+
+def handle_signal(signo: int, _frame: t.Optional[FrameType]) -> None:
+    """Helper function to ensure clean process termination"""
+    if not signo:
+        logger.warning("Received signal with no signo")
+
+    cleanup()
+
+
+def register_signal_handlers() -> None:
+    """Register a signal handling function for all termination events"""
+    for sig in SIGNALS:
+        signal.signal(sig, handle_signal)
+
+
+def get_parser() -> argparse.ArgumentParser:
+    # NOTE: plus prefix avoids passing param incorrectly to python interpreter,
+    # e.g. `python -m smartsim._core.entrypoints.indirect -c ... -t ...`
```

Comment:
Even more pedantic than usual!: `-c` and `-t` are no longer args for the indirect. Still got the point of the message, but my brain is angry they do not match :(

Reply:
removing the comment altogether. i didn't like it's face!